src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,183 @@
+        Assertions.assertThrows(InvalidDataException.class,
+                () -> registrationService.register(null),
+                String.format("Should throw %s when user is null", EXPECTED_EXCEPTION));
+    }
+
+    @Test
+    void register_newLogin_Ok() {
+        User user = new User();
+        user.setLogin("User");
+        user.setPassword("123456");
+        user.setAge(21);
+        registrationService.register(user);
+        User actual = storageDao.get(user.getLogin());
+        Assertions.assertEquals(user, actual,
+                "User should be added if user login doesnt exist");
+    }
+
+    @Test
+    void register_existingLogin_notOk() {
+        User userOne = new User();
+        userOne.setAge(19);
+        userOne.setPassword("123456");
+        userOne.setLogin("User");
+        storageDao.add(userOne);
+        User userTwo = new User();
+        userTwo.setAge(19);
+        userTwo.setPassword("123456");
+        userTwo.setLogin("User");
+        Assertions.assertThrows(InvalidDataException
+                        .class, () -> registrationService.register(userTwo),
+                String.format("Should throw %s when login is already exists",
+                        EXPECTED_EXCEPTION));
+    }
+
+    @Test
+    void register_nullLogin_notOk() {
+        User user = new User();
+        user.setAge(19);
+        user.setPassword("123456");
+        user.setLogin(null);
+        Assertions.assertThrows(InvalidDataException.class,
+                () -> registrationService.register(user),
+                String.format("Should throw %s when login is null", EXPECTED_EXCEPTION));
+    }
+
+    @Test
+    void register_over18Age_Ok() {
+        User user = new User();
+        user.setLogin("User");
+        user.setPassword("123456");
+        user.setAge(21);
+        registrationService.register(user);
+        User actual = storageDao.get(user.getLogin());
+        Assertions.assertEquals(user, actual,
+                "User should be added if user age is valid");
+    }
+
+    @Test
+    void register_nullAge_notOk() {
+        User user = new User();
+        user.setLogin("User");
+        user.setPassword("123456");
+        user.setAge(null);
+        Assertions.assertThrows(InvalidDataException.class,
+                () -> registrationService.register(user),
+                String.format("Should throw %s when age is null", EXPECTED_EXCEPTION));
+    }
+
+    @Test
+    void register_minValueIntegerAge_notOk() {
+        User user = new User();
+        user.setLogin("User");
+        user.setPassword("123456");
+        user.setAge(Integer.MIN_VALUE);
+        Assertions.assertThrows(InvalidDataException.class,
+                () -> registrationService.register(user),
+                String.format("Should throw %s when age "
+                        + "is less than %d years", EXPECTED_EXCEPTION, MIN_AGE));
+    }
+
+    @Test
+    void register_maxValueIntegerAge_notOk() {
```

Comment:
```suggestion
    void register_tooBigAge_notOk() {
```

Reply:
I have deleted method that checked max value so I also deleted this test, fixed